proofs.go

```diff
@@ -42,16 +42,16 @@ func (proof *SparseMerkleProof) sanityCheck(th *treeHasher) bool {
 	}
 
 	// Check that the sibling data hashes to the first side node if not nil
-	if proof.SiblingData != nil {
-		siblingHash := th.digest(proof.SiblingData)
-		if proof.SideNodes != nil && len(proof.SideNodes) > 0 {
-			if !bytes.Equal(proof.SideNodes[0], siblingHash) {
-				return false
-			}
-		}
+	if proof.SiblingData == nil {
+		return true
 	}
 
-	return true
+	if len(proof.SideNodes) == 0 {
```

Comment:
Nit: maybe combine with above condition, so:

```
if proof.SiblingData == nil || len(proof.SideNodes) == 0 {
	return true
}
```

Also maybe write a simple test for this case, so the coverage isn't affected.

Reply:
I've combined the conditions, thanks! For the test, it'll take quite a bit of context and time consuming which I'll delegate to the maintainers of this library :-) I am working on completing our audit currently.